### PR TITLE
fix: 📦 changed way of creating bbox from mask

### PR DESCRIPTION
**Pull request by Yurchikian**
Had some issues while trying to create bbox from mask.

Changed code according to
[This stackoverflow answer](https://stackoverflow.com/questions/31400769/bounding-box-of-numpy-array)

Also fixed processing of mask input

### CODE REVIEW

File: nodes/crop.py [hunk 1]
```diff
@@ -141,19 +130,19 @@ def do_crop(
...
         image = image.numpy()
-        if mask:
-            mask = mask.numpy()
-
         if bbox != None:
             x, y, width, height = bbox
 
+        if mask != None:
+            mask = mask.numpy()
+            cropped_mask = mask[y : y + height, x : x + width]
```
Comment (melMass) [inline nodes/crop.py:138]: this makes cropped_mask unbounded I think (conditionally defined)

**Comment by melMass** [comment]
Thanks a lot for the contribution but I will use a different approach, what happens is I think Comfy changed the way mask are generated  (I could be wrong).
Simply exposing an "invert" input on it seems to fix the issue you were trying to solve.

In case I misdiagnosed your issue can you explain what was not working as expected?

I just pushed the fix I'm mentioning in:
53cb503866da6d83b47eaeb8073039ace2ae0a95

**Comment by Yurchikian** [comment]
I had several issues, that I've tried to fix. One of them is this error:
```
File "Q:\AI\ComfyUI\ComfyUI\custom_nodes\comfy_mtb\nodes\crop.py", line 144, in do_crop
    if mask:
RuntimeError: Boolean value of Tensor with more than one value is ambiguous
```

I believe that you can't just put ndarray into condition. So when the mask is present in Crop node, this error occurs, that's why I've changed it. And it works for both cases when mask is present and when mask is absent. 

What about other issue - I had problems with wrong bbox generation, inverting mask beforehand did not help.
Just checked with updated version - works as expected (except the mentioned problem with mask cropping)

So all changes from my pull request are 
```diff
- if mask:
+ if mask != None:
```

**Review by melMass — APPROVED** [review]
You are right! I removed the other edits that I think are now covered by the invert input, thanks for spotting those issues!

**Comment by melMass** [comment]
I see what you meant for the mask my bad🤦
I'll push some fix and a test workflow now

**Comment by melMass** [comment]
I think it cover all cases, don't hesitate otherwise:

image

**Pull request merged**